### DISCUSSION TITLE
Initial Value for Width/Height/Spacing is a must

### DISCUSSION THREAD

**Discussion by rasan3-1416** [tailwindlabs/tailwindcss#12490] [Ideas]
#Initial value is needed but it's not available

Because the initial value can't be used in width/ height or spacing using the tailwind classes, it is causing issues in design and now for a solution, I have to use the required initial value by configuring the tailwind.config.js and it is really a hassle.

Can't use the initial value. Using the width value 0 is causing code issues
![Screenshot 2023-11-27 143434](https://github.com/tailwindlabs/tailwindcss/assets/83072355/47954cf0-60a5-4d5d-84a1-65a7ca28cc03)

The swiper-button-prev uses the width and height initial value but the swiper-button-next uses 0 as the width and height value which is causing design issues
![Screenshot 2023-11-27 143700](https://github.com/tailwindlabs/tailwindcss/assets/83072355/0e0c902a-fbc2-4399-803d-6846d055a9b8)

I hope the Tailwind community will solve this issue as soon as possible. And here is the hassleful solution I am using.

![Screenshot 2023-11-27 140149](https://github.com/tailwindlabs/tailwindcss/assets/83072355/8e73482b-b549-4284-af6c-696b5b00ce97)

**Comment by adamwathan** [comment]
Hey! You can use `w-[initial]` for this, or in this particular case I would expect simply `w-full` to be the right value as I believe 100% is the initial width of any `div` element.

Not opposed to baking it in to Tailwind by default but feels like a big project because we'd want to assess the whole framework to decide where to add `initial` values and where not.

I don't totally understand your particular example (making the button 0px tall and 0px wide seems odd to me) but if you can create a [Tailwind Play](https://tailwind.new) that has your desired design using `w-initial` I'd be happy to take a look and show you how I'd do it without needing `initial`.
